app/services/content_change_handler_service.rb

```diff
@@ -10,10 +10,12 @@ def self.call(*args)
   end
 
   def call
-    content_change = ContentChange.create!(content_change_params)
-    MetricsService.content_change_created
-    MatchedContentChangeGenerationService.call(content_change: content_change)
-    ProcessContentChangeAndGenerateEmailsWorker.perform_async(content_change.id)
+    ActiveRecord::Base.transaction do
+      content_change = ContentChange.create!(content_change_params)
+      MetricsService.content_change_created
+      MatchedContentChangeGenerationService.call(content_change: content_change)
+      ProcessContentChangeAndGenerateEmailsWorker.perform_async(content_change.id)
```

Comment:
This should happen after the transaction, to ensure that it has committed prior to the job starting.

Reply:
Good catch, I'd changed my mind about it, I was right the first time!